text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,502 @@
+- feat(lib): API Calls/Logic for create/update stack sets
+
+### README
+The ```cdk.StackSet``` construct in CDK defines AWS resources in a stack set and stack set configuration/deployment preferences. This is different from the ```cdk.CfnStackSet``` construct which creates a CFN stack that deploys a CFN stack set. The ```cdk.StackSet``` construct creates the stack set directly if it does not exist, or updates a stack set and existing stack set instances while monitoring the stack set operation.
+
+#### Usage
+To define a stack set, extend the core construct `StackSet`. Then define AWS resources in the same way as defining resources for stacks. Compared to the `Stack` construct, the `StackSet` core construct has different props specific to StackSets. The props define stack set configurations and operation preferences for when updates occur.
+
+**Example:**
+```ts
+class POCStackSet extends cdk.StackSet {
+    constructor(scope: cdk.App, id: string, props: cdk.StackSetProps = {}) {
+        super(scope, id, props);
+
+        new Role(this, 'ExampleRole', {
+          assumedBy: new ServicePrincipal('s3.amazonaws.com')
+        });
+    }
+}
+new POCStackSet(app, 'poc-stack-set',
+    {
+        faultTolerancePercentage: 10,
+        maxConcurrentPercentage: 10,
+        regionConcurrencyType: 'PARALLEL'
+        .
+        .
+        .
+     }
+);
+```
+
+**Deployment:**
+
+To deploy the stack set, the CLI command remains the same: ```cdk deploy```. If the stack set **does not exist**, then CDK will create the stack set directly in CFN. If the stack set **does exist**, then the stack set itself will be updated as well as all **existing** stack set instances. Stack sets instances are added to the stack set outside of the CDK context.
+
+## FAQ
+
+### What are we launching today?
+The core construct StackSet, which will enable users to define and deploy stack sets directly from CDK.
+
+### Why should I use this feature?
+This core constuct directly creates a CFN stack set directly vs. provisioning a single stack that deploys a stack set. The deployment also updates the stack set, updates existing stack set instances, and monitors the stack set operation.
+
+### What is supported?
+* Self Manged Stack Sets
+  * Create, Update, Update Existing Stack Set Instances
+* File Assets Management
+
+### What is not supported?
+* Service Managed Stack Sets
+* Stack Set Instance Add/Remove through CDK
+* Stack Set deletion
+* Docker Assets
+
+## Internal FAQ
+### Why are we doing this?
+CDK does not support the creation and updating of CFN stack sets directly, only through an intermediate single stack. This would also enable customers with existing stack sets to utlize CDK to update in the future.
+
+### Why should we _not_ do this?
+CDK does provide the ```cdk.CFNStackSet``` construct, but requires an intermediate single stack to deploy the stack set which does not monitor stack set operations on update.
```

Comment:
Can you elaborate on this? I am not sure I fully understand the limitations of the L1 and why this is the only way to solve them

Reply:
Good question! Our original use case was to utilize CDK to deploy to existing stack sets, and eventually to create new stack sets in the future. We originally evaluated CFNStackSet, but the way it worked was by deploying a stack with "AWS::CloudFormation::StackSet" CFN resource that deployed the stack set. We found it a little odd where we found it  to require a stack as a wrapper around stack set.

Originally we thought that importing an existing stack set into a new stack was not supported, but I just did some more testing with CFN resource imports and it does work. The caveat is, we would need to pass parameters/tags from the stack to the stack set, migrate all our stack sets to provision from stack (for our team that is upwards of 5000 stack sets), and update all our existing systems to interact with stack sets through stacks vs directly through stack sets. This migration might not be viable for our team.

I looked through a bunch of related git issues regarding stack sets, do you know what the customer had in mind when asking for "native cdk stack set support"? Trying to gauge if this design aligns with those requests, or if CDK team is recommending CFNStackSet as the pattern for deploying stack sets.